posthog-react-native/src/posthog-rn.ts

```diff
@@ -57,27 +75,26 @@ export class PostHog extends PostHogCore {
 
     // Ensure the async storage has been preloaded (this call is cached)
 
-    const setupFromStorage = (): void => {
+    const setupAsync = async (): Promise<void> => {
+      await this._semiAsyncStorage.preloadAsync()
+
       this.setupBootstrap(options)
 
       // It is possible that the old library was used so we try to get the legacy distinctID
       if (!this._semiAsyncStorage.getItem(PostHogPersistedProperty.AnonymousId)) {
-        getLegacyValues().then((legacyValues) => {
-          if (legacyValues?.distinctId) {
-            this._semiAsyncStorage.setItem(PostHogPersistedProperty.DistinctId, legacyValues.distinctId)
-            this._semiAsyncStorage.setItem(PostHogPersistedProperty.AnonymousId, legacyValues.anonymousId)
-          }
-        })
+        const legacyValues = await getLegacyValues()
+        if (legacyValues?.distinctId) {
+          this._semiAsyncStorage.setItem(PostHogPersistedProperty.DistinctId, legacyValues.distinctId)
+          this._semiAsyncStorage.setItem(PostHogPersistedProperty.AnonymousId, legacyValues.anonymousId)
+        }
       }
-    }
 
-    if (this._semiAsyncStorage.isPreloaded) {
-      setupFromStorage()
-    } else {
-      void this._semiAsyncStorage.preloadAsync().then(() => {
-        setupFromStorage()
-      })
+      if (options?.preloadFeatureFlags !== false) {
+        this.reloadFeatureFlags()
+      }
     }
+
+    void setupAsync()
```

Comment:
How do people know when things are in 'ready' state with react native?

I expect everything to bork till this hasn't fulfilled?

Reply:
Indeed. I added a warning above this block to basically say "hey - it looks like you didn't use initAsync (which you should"